classes/ReferenceFactory.php

```diff
@@ -15,95 +15,126 @@ trait ReferenceFactory
+                if ($configure) {
+                    $configure($builder);
+                }
+
+                return $builder;
             }
         );
 
         return $builder;
     }
 
     /**
-     * @param class-string $local
-     * @param class-string $remote
+     * @param class-string  $local
+     * @param class-string  $remote
+     * @param callable|null $configure
+     * @psalm-param callable(LocalKeyBuilder)|null $configure
      * @return LocalKey
      */
-    public function localKey(string $local, string $remote): LocalKey
+    public function localKey(string $local, string $remote, callable $configure = null): LocalKey
     {
         /** @var LocalKey $reference */
-        $reference = $this->buildLocalKey("{$local}local{$remote}", $local, $remote)->resolve();
+        $reference = $this->buildLocalKey("{$local}local{$remote}", $local, $remote, $configure)->resolve();
 
         return $reference;
     }
 
     /**
-     * @param string       $key
-     * @param class-string $local
-     * @param class-string $remote
+     * @param string        $key
+     * @param class-string  $local
+     * @param class-string  $remote
+     * @param callable|null $configure
+     * @psalm-param callable(LocalKeyBuilder)|null $configure
      * @return LocalKeyBuilder
+     * @deprecated Use localKey() instead
      */
-    public function buildLocalKey(string $key, string $local, string $remote): LocalKeyBuilder
+    public function buildLocalKey(string $key, string $local, string $remote, callable $configure = null): LocalKeyBuilder
     {
         /** @var LocalKeyBuilder $builder */
         $builder = $this->references->get(
             $key,
-            function () use ($local, $remote) {
-                return new LocalKeyBuilder($this->manager(), $local, $remote);
+            function () use ($local, $remote, $configure) {
+                $builder = new LocalKeyBuilder($this->manager(), $local, $remote);
+                if ($configure) {
+                    $configure($builder);
+                }
+
+                return $builder;
             }
         );
 
         return $builder;
     }
 
     /**
-     * @param class-string $local
-     * @param class-string $remote
+     * @param class-string  $local
+     * @param class-string  $remote
+     * @param callable|null $configure
+     * @psalm-param callable(JunctionTableBuilder)|null $configure
      * @return JunctionTable
      */
-    public function junctionTable(string $local, string $remote): JunctionTable
+    public function junctionTable(string $local, string $remote, callable $configure = null): JunctionTable
     {
         /** @var JunctionTable $reference */
-        $reference = $this->buildJunctionTable("{$local}junctionTable{$remote}", $local, $remote)->resolve();
+        $reference = $this->buildJunctionTable("{$local}junctionTable{$remote}", $local, $remote,
+            $configure)->resolve();
```

Comment:
Arguments should be all on the same line or all on their own line.

Reply:
Fixed code format and return result without assignment in between.